src/components/Board.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react'
+import Square from './Square'
+import lines from './winning-conditions'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+
+//Board props
+type BoardType = {
+  squares: string[][]
+  updateHistory: (squares: string[]) => void
+  currTurn: string
+}
+
+//component
+const Board: React.FC<BoardType> = (props: BoardType) => {
+  const [boardState, setBoardState] = useState<string[]>(props.squares[0])
+  const [isWinner, setIsWinner] = useState<boolean>(false)
+
+  /**
+   * Handles updating the sate of the board string array
+   * given a value from a square component
+   * @param id
+   * @param value
+   */
+  function handleClick(id: number, value: string) {
+    let newBoardState: string[] = []
+    newBoardState = [...boardState]
+    newBoardState[id] = value
+
+    setBoardState(newBoardState)
+    props.updateHistory(newBoardState)
+  }
+
+  /**
+   * Handler for returning square components ready to be rendered
+   * @param a
+   * @returns Square components
+   */
+  function createSquares(a: string[]) {
+    const squareComponents = a.map((item, index) => {
+      return (
+        <Square
+          key={index}
+          id={index}
+          currTurn={props.currTurn}
+          value={props.squares[props.squares.length - 1][index]}
+          disabled={boardState[index] ? true : false}
+          handleClick={handleClick}
+        />
+      )
+    })
+    return squareComponents
+  }
+  /**
+   * Check for a winner on each click
+   * @param a
+   */
+  function checkWinner(board: string[]) {
```

Comment:
No need to define this function in your React lifecycle, I'd define it under this component or put it in it's own file in src/utils